### PR TITLE
Improve elaboration of PHP configuration

### PR DESCRIPTION
**Pull request by fakoor**
Revise hierarchy of PHP configuration to describe item list for each purpose rather than duplicating the purpose for each affected configuration item. Fix dead link to latest archived copy of the now deprecated magic_quotes_gpc documention page.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -162,12 +162,14 @@ Depending on your Operating System, as well as version of PHP, you may wish to a
...
-* `safe_mode = off` - (If PHP <= v5.4) Allows for SQL Injection (SQLi) [[safe_mode](https://secure.php.net/manual/en/features.safe-mode.php)]
-* `magic_quotes_gpc = off` - (If PHP <= v5.4) Allows for SQL Injection (SQLi) [[magic_quotes_gpc](https://secure.php.net/manual/en/security.magicquotes.php)]
-* `display_errors = off` - (Optional) Hides PHP warning messages to make it less verbose [[display_errors](https://secure.php.net/manual/en/errorfunc.configuration.php#ini.display-errors)]
+* To allow  Remote File Inclusions (RFI):
+    * `allow_url_include = on` [[allow_url_include](https://secure.php.net/manual/en/filesystem.configuration.php#ini.allow-url-include)]
+    * `allow_url_fopen = on` [[allow_url_fopen](https://secure.php.net/manual/en/filesystem.configuration.php#ini.allow-url-fopen)]
+* To allow for SQL Injection (SQLi), if PHP <= v5.4:
+    * `safe_mode = off`  [[safe_mode](https://secure.php.net/manual/en/features.safe-mode.php)]
+    * `magic_quotes_gpc = off` [[magic_quotes_gpc](https://web.archive.org/web/20210111094526/https://www.php.net/manual/en/security.magicquotes.php)]
+* To optinally reduce verbosity by hiding PHP warning messages:
```
Comment (digininja) [inline README.md:171]: Typo, optionally

File: README.md [hunk 2]
```diff
@@ -162,12 +162,14 @@ Depending on your Operating System, as well as version of PHP, you may wish to a
...
-* `allow_url_fopen = on` -  Allows for Remote File Inclusions (RFI)    [[allow_url_fopen](https://secure.php.net/manual/en/filesystem.configuration.php#ini.allow-url-fopen)]
-* `safe_mode = off` - (If PHP <= v5.4) Allows for SQL Injection (SQLi) [[safe_mode](https://secure.php.net/manual/en/features.safe-mode.php)]
-* `magic_quotes_gpc = off` - (If PHP <= v5.4) Allows for SQL Injection (SQLi) [[magic_quotes_gpc](https://secure.php.net/manual/en/security.magicquotes.php)]
-* `display_errors = off` - (Optional) Hides PHP warning messages to make it less verbose [[display_errors](https://secure.php.net/manual/en/errorfunc.configuration.php#ini.display-errors)]
+* To allow  Remote File Inclusions (RFI):
+    * `allow_url_include = on` [[allow_url_include](https://secure.php.net/manual/en/filesystem.configuration.php#ini.allow-url-include)]
+    * `allow_url_fopen = on` [[allow_url_fopen](https://secure.php.net/manual/en/filesystem.configuration.php#ini.allow-url-fopen)]
+* To allow for SQL Injection (SQLi), if PHP <= v5.4:
```
Comment (digininja) [inline README.md:168]: Drop this section, I don't think we need to support php 5.x any more.

**Comment by digininja** [comment]
Looks good, if you could make a couple of changes I'll get that pulled.

**Comment by digininja** [comment]
Thanks. I had to do the merge on the command line for some reason so I hope that you got the credit for the work in Github.

Out of interest, what are you using DVWA for? I assume learning or teaching if you are interested enough to be submitting changes.

**Comment by fakoor** [comment]
Thank you so much. 

You are right, some sorts of learning/teaching scenario is involved. Actually I was queried about DVWA by a friend of mine who is taking some security courses.  She was interested to see that the README  was available in a couple of other languages and was wondering why no one has ever submitted a Persian translation for this. I got interested in and was motivated to do so. That is mainly a passion for writing and contributing to improving things.

**Comment by digininja** [comment]
All contributions are very welcome thanks.

**Pull request merged**